code/game/objects/items/stacks/golem_food/golem_status_effects.dm

```diff
@@ -362,28 +432,26 @@
 	victim.apply_damage(mining_bonus, BRUTE)
 
 /// Make the targeted arm big and strong
-/datum/status_effect/golem/titanium/proc/buff_arm(obj/item/bodypart/arm/arm)
-	arm.unarmed_damage_low += damage_increase
-	arm.unarmed_damage_high += damage_increase
-	RegisterSignal(arm, COMSIG_QDELETING, PROC_REF(on_arm_destroyed))
-	LAZYADD(modified_arms, arm)
+/datum/status_effect/golem/titanium/set_arm_effects()
+	. = ..()
+	for(var/obj/item/bodypart/arm/arm as anything in modified_arms)
+		arm.unarmed_damage_low += damage_increase
+		arm.unarmed_damage_high += damage_increase
+		RegisterSignal(arm, COMSIG_QDELETING, PROC_REF(on_arm_destroyed))
 
 /datum/status_effect/golem/titanium/on_remove()
-	var/mob/living/carbon/human/human_owner = owner
 	UnregisterSignal(human_owner, COMSIG_LIVING_UNARMED_ATTACK)
 	human_owner.physiology.brute_mod /= brute_modifier
-	for (var/obj/item/bodypart/arm/arm as anything in modified_arms)
-		debuff_arm(arm)
-	LAZYCLEARLIST(modified_arms)
+	reset_arm_effects()
 	return ..()
 
 /// Make the targeted arm small and weak
-/datum/status_effect/golem/titanium/proc/debuff_arm(obj/item/bodypart/arm/arm)
-	if (!arm)
-		return
-	arm.unarmed_damage_low -= damage_increase
-	arm.unarmed_damage_high -= damage_increase
-	UnregisterSignal(arm, COMSIG_QDELETING)
+/datum/status_effect/golem/titanium/reset_arm_effects()
+	for(var/obj/item/bodypart/arm/arm as anything in modified_arms)
+		arm.unarmed_damage_low -= damage_increase
+		arm.unarmed_damage_high -= damage_increase
+		UnregisterSignal(arm, COMSIG_QDELETING)
+	. = ..()
```

Comment:
Some people prefer `return ..()` if it's the end of a proc anyway, though I'm not enforcing this as a requirement.